### PR TITLE
fix(amf): Addressing the warnings during compilation

### PR DESCRIPTION
**Pull request by RahulKalsangra**
Signed-off-by: RahulKalsangra <rahul.kalsangra@wavelabs.ai>

Remove unused variable from files to avoid warning
cast some variable for suitable comparison
## Test Plan
Capture Packet

packet10401

TestLog

test10401

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by panyogesh — COMMENTED** [review]
To start with lets focus only on the AMF and NGAP changes.

File: lte/cloud/go/protos/oai/sgw_state.pb.go [hunk 1]
```diff
@@ -243,38 +243,37 @@ func init() {
...
-	0x67, 0x76, 0xaf, 0x78, 0x8e, 0x9e, 0xd5, 0xce, 0x9b, 0x45, 0xd3, 0xbb, 0xeb, 0xbd, 0x07, 0xde,
-	0x7b, 0xb3, 0xd8, 0x75, 0x1f, 0xdd, 0xa0, 0x6e, 0xca, 0xab, 0x2b, 0xb6, 0xdd, 0x01, 0xfe, 0x8e,
-	0x30, 0xa8, 0x5c, 0xfb, 0x71, 0x40, 0x28, 0x40, 0xc3, 0x20, 0x3a, 0x88, 0x3b, 0xb3, 0xf8, 0xff,
-	0x44, 0xf7, 0x2f, 0x29, 0x79, 0x6c, 0x79, 0xb5, 0x87, 0x8e, 0xbe, 0xa1, 0x43, 0xa0, 0xc3, 0x16,
-	0xf1, 0x1b, 0x74, 0xec, 0x77, 0xe0, 0xd6, 0x4a, 0x31, 0xbf, 0x81, 0xa0, 0x6e, 0x12, 0xd0, 0x4b,
-	0x0f, 0xae, 0x28, 0x1e, 0xa2, 0x0e, 0x77, 0x66, 0x33, 0x5d, 0xfb, 0xce, 0xb7, 0x7b, 0x43, 0x35,
-	0x04, 0x7d, 0x9f, 0x0f, 0x7f, 0xbc, 0xf0, 0x79, 0x26, 0xb0, 0xfa, 0xbc, 0xd0, 0x6b, 0x3a, 0xe1,
-	0x7a, 0x67, 0xfe, 0xd7, 0x6d, 0x7f, 0x3e, 0xfb, 0x1b, 0x00, 0x00, 0xff, 0xff, 0x98, 0x04, 0x3c,
-	0x3b, 0x8e, 0x03, 0x00, 0x00,
+	// 510 bytes of a gzipped FileDescriptorProto
```
Comment (panyogesh) [inline lte/cloud/go/protos/oai/sgw_state.pb.go]: Lets not start with proto changes of 4G in this PR. If required lets address in separate PR.

File: lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/include/NwGtpv2c.h [hunk 2]
```diff
@@ -644,7 +644,7 @@ nw_rc_t nwGtpv2cProcessUlpReq(
  @param[in] arg : Pointer timeout arguments.
  @return NW_OK on success.
  */
-nw_rc_t nwGtpv2cProcessTimeoutExt(
+int nwGtpv2cProcessTimeoutExt(
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/include/NwGtpv2c.h]: Lets keep GTP changes outside this PR.

File: lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/src/NwGtpv2c.c [hunk 3]
```diff
@@ -2171,7 +2171,7 @@ nw_rc_t nwGtpv2cProcessTimeoutOld(void* arg) {
   OAILOG_FUNC_RETURN(LOG_GTPV2C, rc);
 }
 
-nw_rc_t nwGtpv2cProcessTimeoutExt(zloop_t* loop, int timer_id, void* arg) {
+int nwGtpv2cProcessTimeoutExt(zloop_t* loop, int timer_id, void* arg) {
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/src/NwGtpv2c.c]: Same as above.

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp [hunk 4]
```diff
@@ -212,17 +212,10 @@ void clear_amf_smf_context(std::shared_ptr<smf_context_t> smf_ctx) {
 int pdu_session_release_request_process(
     ue_m5gmm_context_s* ue_context, std::shared_ptr<smf_context_t> smf_ctx,
     amf_ue_ngap_id_t amf_ue_ngap_id, bool retransmit) {
-  int rc                = 1;
-  amf_smf_t amf_smf_msg = {};
+  int rc = 1;
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp]: replace "1" with "RETURNerror"

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp [hunk 5]
```diff
@@ -212,17 +212,10 @@ void clear_amf_smf_context(std::shared_ptr<smf_context_t> smf_ctx) {
...
   // amf_cause = amf_smf_handle_pdu_release_request(
   //              msg, &amf_smf_msg);
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp]: Remove the commented code (if not applicable).

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.c [hunk 6]
```diff
@@ -50,16 +50,17 @@ const struct gtp_tunnel_ops* upf_gtp_tunnel_ops_init_openflow(void) {
 
 int upf_add_tunnel(
     struct in_addr ue, struct in6_addr* ue_ipv6, int vlan, struct in_addr enb,
-    uint32_t i_tei, uint32_t o_tei, Imsi_t imsi, struct ip_flow_dl* flow_dl,
-    uint32_t flow_precedence_dl, char* apn) {
+    struct in6_addr* enb_ipv6, uint32_t i_tei, uint32_t o_tei, Imsi_t imsi,
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.c]: Lets have these changes outside this PR.

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.h [hunk 7]
```diff
@@ -23,12 +23,13 @@ const struct gtp_tunnel_ops* upf_gtp_tunnel_ops_init_openflow(void);
 
 int upf_add_tunnel(
     struct in_addr ue, struct in6_addr* ue_ipv6, int vlan, struct in_addr enb,
-    uint32_t i_tei, uint32_t o_tei, Imsi_t imsi, struct ip_flow_dl* flow_dl,
-    uint32_t flow_precedence_dl, char* apn);
+    struct in6_addr* enb_ipv6, uint32_t i_tei, uint32_t o_tei, Imsi_t imsi,
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.h]: Same as above.

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp [hunk 8]
```diff
@@ -24,11 +24,15 @@ void mme_app_log_ipv4_imsi_map() {
   OAILOG_FUNC_IN(LOG_MME_APP);
   UeIpImsiMap& ueip_imsi_map =
       MmeNasStateManager::getInstance().get_mme_ueip_imsi_map();
-  for (const auto& itr_map : ueip_imsi_map) {
-    for (const auto& it_vec : itr_map.second) {
+
+  for (std::unordered_map<std::string, std::vector<uint64_t>>::iterator
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp]: Same as above.

File: lte/gateway/c/core/oai/tasks/nas/emm/Attach.c [hunk 9]
```diff
@@ -1072,8 +1072,8 @@ status_code_e _emm_attach_reject(
     emm_as_set_security_data(
         &emm_sap.u.emm_as.u.establish.sctx, NULL, false, false);
   }
-  rc = emm_sap_send(&emm_sap);
-  attach_proc->attach_reject_sent++;
+  rc                              = emm_sap_send(&emm_sap);
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/nas/emm/Attach.c]: Same as above.

File: lte/protos/oai/sgw_state.proto [hunk 10]
```diff
@@ -14,7 +14,6 @@ limitations under the License.
 syntax = "proto3";
 
 import "lte/protos/oai/common_types.proto";
```
Comment (panyogesh) [inline lte/protos/oai/sgw_state.proto]: Lets keep proto changes outside.

**Comment by panyogesh** [comment]
Kindly attach the 5G working test logs.

File: lte/protos/oai/sgw_state.proto (hunk 10 above)
Comment (RahulKalsangra) [reply lte/protos/oai/sgw_state.proto]: Comment addressed

File: lte/gateway/c/core/oai/tasks/nas/emm/Attach.c (hunk 9 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/nas/emm/Attach.c]: Comment addressed

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp (hunk 8 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp]: Comment addressed

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.h (hunk 7 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.h]: Comment addressed

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.c (hunk 6 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.c]: Comment addressed

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp (hunk 5 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp]: Comment addressed

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp (hunk 4 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp]: Comment addressed

File: lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/src/NwGtpv2c.c (hunk 3 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/src/NwGtpv2c.c]: Comment addressed

File: lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/include/NwGtpv2c.h (hunk 2 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/include/NwGtpv2c.h]: Comment addressed

File: lte/cloud/go/protos/oai/sgw_state.pb.go (hunk 1 above)
Comment (RahulKalsangra) [reply lte/cloud/go/protos/oai/sgw_state.pb.go]: Comment addressed

**Comment by RahulKalsangra** [comment]
> Kindly attach the 5G working test logs.

test logs Attached

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c [hunk 11]
```diff
@@ -565,7 +565,7 @@ status_code_e ngap_generate_downlink_nas_transport(
 /*ngap initial context setup request message to gNB*/
 void ngap_handle_conn_est_cnf(
     ngap_state_t* state,
-    const Ngap_initial_context_setup_request_t* const conn_est_cnf_pP) {
+    Ngap_initial_context_setup_request_t* conn_est_cnf_pP) {
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c]:
```suggestion
    Ngap_initial_context_setup_request_t* const conn_est_cnf_pP) {
```

**Review by ssanadhya — APPROVED** [review]
Thanks for the cleanup! Left a minor comment. Rest looks good to me. Please try to fix the cpplint warning too.

**Review by ulaskozat — APPROVED** [review]
LGTM

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c (hunk 11 above)
Comment (RahulKalsangra) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c]: "ngap_amf_nas_procedures.c" Change Addressed . For cpplint Actually tried resolving the cpp-lint error but it seems "ngap_amf_handlers.c" is being treated as c object hence it says reinterpret_cast undeclared. While changing this to cpp file it's giving more errors as this is dependent on NGAP auto-generated file.

Error Message 
Screenshot from 2021-11-25 10-44-55


log:
[log.txt](https://github.com/magma/magma/files/7602588/log.txt)

**Pull request merged**